### PR TITLE
Adding support for global secondary indexes

### PR DESCRIPTION
**Pull request by nhibberd**
! @charleso @erikd-ambiata @jystic 
/jury approved @charleso @erikd-ambiata

### CODE REVIEW

**Comment by charleso** [comment]
👎

**Comment by charleso** [comment]
👍

File: src/Spine/Data.hs [hunk 1]
```diff
@@ -64,7 +68,27 @@ newtype TableName =
       renderTableName :: Text
     } deriving (Eq, Show)
 
+newtype IndexName =
+  IndexName {
+      renderIndexName :: Text
+    } deriving (Eq, Show)
+
+data SecondaryIndex = forall a.
```
Comment (erikd-ambiata) [inline src/Spine/Data.hs]: Wow, `ExistentialQuantification`. Might be nice to have a small comment about what and why.

**Comment by erikd-ambiata** [comment]
👍

File: src/Spine/Data.hs (hunk 1 above)
Comment (nhibberd) [reply src/Spine/Data.hs]: Will raise an issue to come back to do this, probably need a sweep of the entire project for more doc

**Pull request merged**